### DISCUSSION TITLE
🎉 New Feature: DOT Language Support 🎉

### DISCUSSION THREAD

**Discussion by s0l0ist** [vizdom-dev/vizdom#8] [Announcements]
While Vizdom aims to be a code-first approach to diagramming, I'm excited to share that Vizdom now has partial support for the [DOT language](https://graphviz.org/doc/info/lang.html), a widely used format for defining graphs and is what powers the Live [Editor](https://vizdom.dev/editor/view) and [Diff](https://vizdom.dev/editor/diff) views.

# Key DOT Features:
 - **Cluster Graph Support**: Vizdom now parses [cluster graphs](https://graphviz.org/docs/attrs/cluster/) (e.g., subgraphs with the `cluster_` prefix or `cluster=true` attributes).
 - **Styling Attributes**: Most DOT styling attributes are supported right away.
 - **Graceful Fallbacks**: Unsupported styles are ignored or default to safe options for rendering.

## Example

```typescript
import { DotParser } from "@vizdom/vizdom-ts-node";

// Initialize a new DOT parser
const parser = new DotParser();
const dotGraph = parser.parse("digraph { a -> b }");
const directedGraph = dotGraph.to_directed();
const positioned = directedGraph.layout();

// Export to SVG
await fs.writeFile("./graph.svg", positioned.to_svg().to_string());
```
**NOTE**: **undirected** diagrams aren’t supported yet, as Vizdom’s layout engine is currently focused on directed graphs.

For more information, checkout the [dot example](https://github.com/vizdom-dev/vizdom/blob/main/examples/dot/index.ts)

**Comment by gwennlbh** [comment]
The link to the dot example is broken, i think you meant https://github.com/vizdom-dev/vizdom/tree/main/examples/dot

Really cool new feature though, we might be able to drop a system dependency in our dev setup with this !

**Reply by s0l0ist** [reply]
oh lol, thanks!

**Reply by s0l0ist** [reply]
> Really cool new feature though, we might be able to drop a system dependency in our dev setup with this !

Let me know if you run into any issues. I'd be happy to help!
